instance/models/server.py

```diff
@@ -316,18 +319,23 @@ def update_status(self):
 
         # First check if it makes sense to update the current status.
         # This is not the case if we can not interact with the server:
-        if self.status == Status.BuildFailed:
+        if self.status in (Status.BuildFailed, Status.Terminated):
+            return self.status
+        try:
+            os_server = self.os_server
+        except (requests.RequestException, novaclient.exceptions.ClientException):
+            self.logger.debug('Could not reach the OpenStack API')
```

Comment:
Your latest code change fixed my GUI issue, so after the requests timeout, I am now able to see the Unknown status.